migrator/migrations/m_178_to_m_179_openshift_policy_exclusions/postgres/policy_migrator.go

```diff
@@ -0,0 +1,318 @@
+	"github.com/stackrox/rox/pkg/postgres"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/set"
+)
+
+// PolicyDiff is an alternative to PolicyChanges that automatically constructs migrations based on diffs of policies.
+type PolicyDiff struct {
+	FieldsToCompare []FieldComparator
+	PolicyFileName  string
+}
+
+// PolicyChanges lists the fields that must match before a policy is updated and what it should be updated to
+type PolicyChanges struct {
+	FieldsToCompare []FieldComparator
+	// ToChange is the set of changes that must be made to this policy
+	ToChange PolicyUpdates
+}
+
+// PolicyUpdates lists the possible fields of a policy that can be updated. Any nil fields will not be updated
+// In order to change an item in an array (e.g. exclusions), remove the existing one and add the updated one back in.
+type PolicyUpdates struct {
+	// PolicySections is the new policy sections
+	PolicySections []*storage.PolicySection
+	// MitreVectors is the new MITRE ATT&CK section
+	MitreVectors []*storage.Policy_MitreAttackVectors
+	// ExclusionsToAdd is a list of exclusions to insert (or append) to policy
+	ExclusionsToAdd []*storage.Exclusion
+	// ExclusionsToRemove is a list of exclusions to remove from policy
+	ExclusionsToRemove []*storage.Exclusion
+	// Name is the new name for the policy
+	Name *string
+	// Remediation is the new remediation string
+	Remediation *string
+	// Rationale is the new rationale string
+	Rationale *string
+	// Description is the new description string
+	Description *string
+	// Disable is true if the policy should be disabled, false if it should be enabled and nil for no change
+	Disable *bool
+}
+
+// FieldComparator should compare policies and return true if they match for a defined field
+type FieldComparator func(first, second *storage.Policy) bool
+
+// ReadPolicyFromFile reads policies from file given the path and the collection of files.
+func ReadPolicyFromFile(fs embed.FS, filePath string) (*storage.Policy, error) {
+	contents, err := fs.ReadFile(filePath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to read file %s", filePath)
+	}
+	var policy storage.Policy
+	err = jsonutil.JSONBytesToProto(contents, &policy)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to unmarshal policy json at path %s", filePath)
+	}
+	return &policy, nil
+}
+
+func diffPolicies(beforePolicy, afterPolicy *storage.Policy) (PolicyUpdates, error) {
+	if beforePolicy == nil {
+		return PolicyUpdates{}, errors.New("policy to be migrated must not be nil")
+	}
+	if afterPolicy == nil {
+		return PolicyUpdates{}, errors.New("policy to be migrated does not have valid target, but found nil target")
+	}
+
+	// Clone policies because we mutate them.
+	beforePolicy = beforePolicy.Clone()
+	afterPolicy = afterPolicy.Clone()
+
+	var updates PolicyUpdates
+
+	// Diff exclusions and clear out if they are similar
+	getExclusionsUpdates(beforePolicy, afterPolicy, &updates)
+	beforePolicy.Exclusions = nil
```

Comment:
nit:  Could we do all the updates the clear out the policy in a function or all at once at the end?  To me that would be cleaner than doing it after each `if` block